core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/AutoFeatureAugmentor.java

```diff
@@ -0,0 +1,7 @@
+package io.quarkus.deployment.builditem.nativeimage;
+
+import io.quarkus.gizmo.TryBlock;
+
+public interface AutoFeatureAugmentor {
+    void augment(TryBlock overallCatch);
```

Comment:
I think you should pass a `BytecodeCreator` here instead. That would be more future proof.